Parsing error when parserOptions.ecmaVersion is not set

### Initial checklist

- [X] I read the [support docs](https://mdxjs.com/community/support/)
- [X] I read the [contributing guide](https://mdxjs.com/community/contribute/)
- [X] I agree to follow the [code of conduct](https://github.com/mdx-js/.github/blob/HEAD/code-of-conduct.md)
- [X] I searched [issues](https://github.com/search?q=user%3Amdx-js&type=Issues) and couldn’t find anything (or linked relevant results below)

### Affected packages and versions

latest

### Link to runnable example

_No response_

### Steps to reproduce

A very basic example following the instructions within this repo is already failing for me with a parsing error:
```
  0:0  error  Parsing error: ImportDeclaration should appear when the mode is ES6 and in the module context
```

I've reduced my `.eslintrc.json` to the simplest possible.

```json
{
  "plugins": ["@typescript-eslint"],
  "extends": [
    "next/core-web-vitals",
    "plugin:prettier/recommended",
    "plugin:mdx/recommended"
  ],
  "settings": {
    "mdx/code-blocks": true,
    "mdx/language-mapper": {}
  }
}
```

When I add the following to the `.eslintrc.json` it works as expected
```json
  "parserOptions": {
    "ecmaVersion": 12
  }
```

If this isn't a bug, it should be documented somewhere. It took a lot of time to figure this out.

### Expected behavior

My MDX file should be linted without errors

### Actual behavior

```0:0  error  Parsing error: ImportDeclaration should appear when the mode is ES6 and in the module context```

### Runtime

Node v16

### Package manager

yarn v2, yarn v1

### OS

macOS

### Build and bundle tools

Next.js

--- a/packages/eslint-mdx/src/parser.ts
+++ b/packages/eslint-mdx/src/parser.ts
@@ -45,8 +45,7 @@ export const DEFAULT_PARSER_OPTIONS: ParserOptions = {
   ecmaFeatures: {
     jsx: true,
   },
-  ecmaVersion:
-    new Date().getUTCFullYear() as Linter.ParserOptions['ecmaVersion'],
+  ecmaVersion: 2015,
   sourceType: 'module',
   tokens: true,
   filePath: PLACEHOLDER_FILE_PATH,

